crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -564,7 +565,7 @@ impl Inner {
             let table_row = StTableRow::try_from(&row)?;
             let table_id = TableId(table_row.table_id);
             let schema = self.schema_for_table(table_id)?;
-            let row_type = self.row_type_for_table(table_id)?;
+            let row_type = self.row_type_for_table(table_id)?.into_owned();
```

Comment:
I'd echo Ingvar's comments. I think eventually we'll converge on the right patterns uniformally throughout this file, though. I don't think we need to do all of it at once atomically.